flink-ml-lib/src/main/java/org/apache/flink/ml/feature/robustscaler/RobustScalerModel.java

```diff
@@ -0,0 +1,179 @@
+                (StreamTableEnvironment) ((TableImpl) inputs[0]).getTableEnvironment();
+        DataStream<Row> inputStream = tEnv.toDataStream(inputs[0]);
+
+        RowTypeInfo inputTypeInfo = TableUtils.getRowTypeInfo(inputs[0].getResolvedSchema());
+        RowTypeInfo outputTypeInfo =
+                new RowTypeInfo(
+                        ArrayUtils.addAll(inputTypeInfo.getFieldTypes(), VectorTypeInfo.INSTANCE),
+                        ArrayUtils.addAll(inputTypeInfo.getFieldNames(), getOutputCol()));
+        final String broadcastModelKey = "broadcastModelKey";
+        DataStream<RobustScalerModelData> modelDataStream =
+                RobustScalerModelData.getModelDataStream(modelDataTable);
+
+        DataStream<Row> output =
+                BroadcastUtils.withBroadcastStream(
+                        Collections.singletonList(inputStream),
+                        Collections.singletonMap(broadcastModelKey, modelDataStream),
+                        inputList -> {
+                            DataStream inputData = inputList.get(0);
+                            return inputData.map(
+                                    new PredictOutputFunction(
+                                            broadcastModelKey,
+                                            getInputCol(),
+                                            getWithCentering(),
+                                            getWithScaling()),
+                                    outputTypeInfo);
+                        });
+
+        return new Table[] {tEnv.fromDataStream(output)};
+    }
+
+    /** This operator loads model data and predicts result. */
+    private static class PredictOutputFunction extends RichMapFunction<Row, Row> {
+        private final String broadcastModelKey;
+        private final String inputCol;
+        private final boolean withCentering;
+        private final boolean withScaling;
+
+        private DenseVector medians;
+        private DenseVector scales;
+
+        public PredictOutputFunction(
+                String broadcastModelKey,
+                String inputCol,
+                boolean withCentering,
+                boolean withScaling) {
+            this.broadcastModelKey = broadcastModelKey;
+            this.inputCol = inputCol;
+            this.withCentering = withCentering;
+            this.withScaling = withScaling;
+        }
+
+        @Override
+        public Row map(Row row) throws Exception {
+            if (medians == null) {
+                RobustScalerModelData modelData =
+                        (RobustScalerModelData)
+                                getRuntimeContext().getBroadcastVariable(broadcastModelKey).get(0);
+                medians = modelData.medians;
+                scales =
+                        new DenseVector(
+                                Arrays.stream(modelData.ranges.values)
+                                        .map(range -> range == 0 ? 1 : 1 / range)
```

Comment:
Could you please explain why this PR chooses to make scale  = 1 when range == 0? Spark seems to make scale = 0 in this case.

Reply:
IMO, if set scale = 1, the model would do nothing to the original data which is chosen by sklearn, but if set scale = 0, all records would be replaced by 0 which makes this feature useless.